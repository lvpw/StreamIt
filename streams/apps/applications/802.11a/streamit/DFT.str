/* extremely simple N^2 algorithm for calculating N point DFT */

complex->complex splitjoin DFT(int N) {

  split duplicate;

  for(int i=0; i<N; i++)
    add calc_kth_coeff(N,i); 

  join roundrobin(1);
}


complex->complex pipeline IDFT(int N) {

  add complex_conjugate();
  add DFT(N);
  add complex_conjugate();
  add divide_by_N(N);

}


complex->complex pipeline calc_kth_coeff(int N, int k) {

  add splitjoin {
    split roundrobin(N,0);
    add Identity<complex>;
    add create_unity_train(N,k);
    join roundrobin(1,1);
  }

  add mult_and_sum(N);

} 



complex->complex filter mult_and_sum(int N) {

  work push 1 pop 2*N {

    complex total = 0.0;

    for(int i=0; i<N; i++)
      total += pop()*pop();

    push(total);

  }
}


void->complex filter create_unity_train(int N, int k) {

  complex temp;

  init {
    complex num = 0.0 - 1.0i;
    temp = 2*pi*num*k/N;
  }
  
  work push N pop 0 {

    for(int i=0; i<N; i++) 
      push(exp(temp*i));

  }
}


complex->complex filter divide_by_N(int len) {

  work push len pop len {

    for(int i=0; i<len; i++)
      push(pop()/len);
  }
}



complex->complex filter complex_conjugate() {

  work push 1 pop 1 {

    complex c = pop();
    c.imag = -c.imag;
    push(c);
  }
}
