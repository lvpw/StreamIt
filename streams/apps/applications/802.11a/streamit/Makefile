# import the FFT without the void->void section
all: ../../../benchmarks/fft/streamit/FFT6.str
	./importFFT.pl > FFT6.str
