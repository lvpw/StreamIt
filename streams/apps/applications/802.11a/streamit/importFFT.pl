#!/usr/local/bin/perl

# Writes out all of FFT6 except for the void->void stream.  We need to
# do this since strc expects only one void->void stream in the input
# files.

open(MYFILE, "../../../benchmarks/fft/streamit/FFT6.str") or die("Cannot find FFT6.str.\n");
{
    local $/;
    $text = <MYFILE>;
}

$text =~ s!void->void.*complex->complex filter CombineDFT!complex->complex filter CombineDFT!is;

print $text;
close(MYFILE);
