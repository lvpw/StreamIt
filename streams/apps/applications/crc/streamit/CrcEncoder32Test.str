/**
 *   CRCEncoder32:
 *   CRC Encoder using a 32 bit
 *   Generator Polynomial
 *   NOTE:  Until finite input lengths are accepted by StreamIt,
 *          this will not completely work as expected.
 *   - J. Wong
 *
 *  (Bill translated to new syntax, converted state to peeking)
 */

int->int filter AdditionFilter {
    work push 2 pop 2 {
	int mInitFeedbackData = pop();
	int mShiftRegisterInput = pop();
        int mAdditionOutput;
	
	if(((mInitFeedbackData == 1) && (mShiftRegisterInput == 1)) || ((mInitFeedbackData == 0) && (mShiftRegisterInput == 0)))
	{
	    mAdditionOutput = 0;
	}
	else
	{
	    mAdditionOutput = 1;    
	} 
	push (mInitFeedbackData);
	push (mAdditionOutput);
    }
}

int->int filter InitialAdditionFilter {
    work push 2 pop 2 {
	int mFileInput = pop();
	int mInitFeedbackData = pop();
        int mAdditionOutput;
	//System.err.println("File Input is " + mFileInput + " Feedback data is " + mInitFeedbackData);
	if(((mInitFeedbackData == 1) && (mFileInput == 1)) || ((mInitFeedbackData == 0) && (mFileInput == 0))) {
	    mAdditionOutput = 0;
	}
	else {
	    mAdditionOutput = 1;    
	} 
	push(mAdditionOutput);
	push(mAdditionOutput);
    }

}

/*
int->int filter ShiftRegisterFilter() {
    int mRegisterContents;
    
    work push 2 pop 2 {
	int mInitFeedbackData = pop();
	int mDataStream = pop();
	int mRegisterOutput = mRegisterContents;   //move datastream value into Shift Register
// 	mRegisterContents = mDataStream;       //move original contents to output
// 	mTerminationCounter++;
// 	if(mTerminationCounter < EncoderInput.mInputLength + 1)
// 	{
// 	    System.err.println("Shift Register Contents of Filter " + mRegisterNumber + " at iteration " + mTerminationCounter + " is " + mRegisterContents);
// 	    System.err.println("Shift Register Output of Filter " + mRegisterNumber + " at iteration " + mTerminationCounter + " is " + mRegisterOutput);
// 	}
	push(mInitFeedbackData);
	push(mRegisterOutput);
    }
}
*/

// state eliminated in previous filter in favor of peeking
int->int filter ShiftRegisterFilter() {
    prework push 2 pop 1 {
	push(pop());
	push(0);
    }
    
    work push 2 pop 2 {
	int mRegisterContents = pop();
	int mInitFeedbackData = pop();
	push(mInitFeedbackData);
	push(mRegisterContents);
    }
}

void->int filter CrcInputFilter {
    // int[] mdata;
    boolean counter;
    // EncoderInput mencodefile;

    work push 1 {
	if(!counter)
	{
	    counter = true;
	    push (0);
	}
	else
	{
	    push (1);
	}     
    }
}

int->int filter FeedbackEndFilter {
    work pop 2 push 1 {
	int mInitFeedbackData = pop();
	int mRegisterOutput = pop();
	//System.err.println("Shift Register 3 is " + mRegisterOutput);
	push(mRegisterOutput);
    }
}

int->void filter IntPrinter {
    work pop 1 {
	println(pop());
    }
}

int->int feedbackloop CrcFeedbackLoop {
    enqueue(0);
    join roundrobin();
    body FeedbackBodyStream();
    split duplicate;
    loop Identity<int>();
}

int->int pipeline FeedbackBodyStream {
    add InitialAdditionFilter();
    add ShiftRegisterFilter();
    add AdditionFilter();
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^2
    add ShiftRegisterFilter();
    //x^3
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^4
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^5
    add ShiftRegisterFilter();
    //x^6
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^7
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^8
    add ShiftRegisterFilter();
    //x^9
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^10
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^11
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^12
    add ShiftRegisterFilter();
    //x^13
    add ShiftRegisterFilter();
    //x^14
    add ShiftRegisterFilter();
    //x^15
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^16
    add ShiftRegisterFilter();
    //x^17
    add ShiftRegisterFilter();
    //x^18
    add ShiftRegisterFilter();
    //x^19
    add ShiftRegisterFilter();
    //x^20
    add ShiftRegisterFilter();
    //x^21
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^22
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^23
    add ShiftRegisterFilter();
    //x^24
    add ShiftRegisterFilter();
    //x^25
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^26
    add ShiftRegisterFilter();
    //x^27
    add ShiftRegisterFilter();
    //x^28
    add ShiftRegisterFilter();
    //x^29
    add ShiftRegisterFilter();
    //x^30
    add ShiftRegisterFilter();
    //x^31
    add Identity<int>();

    //add ShiftRegisterFilter(3);
    add FeedbackEndFilter();
}

void->void pipeline CrcEncoder32Test {
    add CrcInputFilter();
    add CrcFeedbackLoop();
    add IntPrinter();
}
