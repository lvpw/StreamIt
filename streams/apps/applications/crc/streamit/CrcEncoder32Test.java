/**
 *   CRCEncoder32.java:
 *   CRC Encoder using a 32 bit
 *   Generator Polynomial
 *   NOTE:  Until finite input lengths are accepted by StreamIt,
 *          this will not completely work as expected.
 *   - J. Wong
 */

import streamit.library.*;
import java.lang.*;
import streamit.library.io.*;
import java.io.*;

/*
class EncoderInput
{

    //member variables!
    public static int mInputLength = 5;


    public int[] readFile() 
  {
    int[] input = new int[mInputLength];
    try
    {
	File f1 = new File("CrcEncoderInput1");
	java.io.FileReader fr = new java.io.FileReader(f1);
	BufferedReader br = new BufferedReader(fr);
	//DataInputStream data = new DataInputStream(new FileInputStream(f1));
	//read the sucker!
	//int[] input = new int[4];
	for(int i = 0; i < input.length; i++)
	    {
		String j = br.readLine();
		
		if (j.equals("1"))
		    {
			input[i] = 1;
		    }
		else
		    { 
			input[i] = 0;
		    }		
		
	    }
	br.close();
    }
    catch (IOException e)
    {
	System.out.println("Blah! IO error!");
    }

    return input;
  }//readFile

}//class EncoderInput
*/

class AdditionFilter extends Filter
{
    //inputs:
    int mInitFeedbackData;
    int mShiftRegisterInput;
    //outputs
    //output 1: mInitFeedbackData
    int mAdditionOutput;


    public void init()
    {
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 2);
    }

    public void work()
    {
	mInitFeedbackData = input.popInt();
	mShiftRegisterInput = input.popInt();
	
	if(((mInitFeedbackData == 1) && (mShiftRegisterInput == 1)) || ((mInitFeedbackData == 0) && (mShiftRegisterInput == 0)))
	{
	    mAdditionOutput = 0;
	}
	else
	{
	    mAdditionOutput = 1;    
	} 
	output.pushInt(mInitFeedbackData);
	output.pushInt(mAdditionOutput);
    }

}//AdditionFilter

class InitialAdditionFilter extends Filter

{
    //inputs:
    int mInitFeedbackData;
    int mFileInput;
    //outputs
    //output 1: mInitFeedbackData
    int mAdditionOutput;


    public void init()
    {
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 2);
    }

    public void work()
    {
	mFileInput = input.popInt();
	mInitFeedbackData = input.popInt();
	//System.err.println("File Input is " + mFileInput + " Feedback data is " + mInitFeedbackData);
	if(((mInitFeedbackData == 1) && (mFileInput == 1)) || ((mInitFeedbackData == 0) && (mFileInput == 0)))
	{
	    mAdditionOutput = 0;
	}
	else
	{
	    mAdditionOutput = 1;    
	} 
	output.pushInt(mAdditionOutput);
	output.pushInt(mAdditionOutput);
    }

}//InitialAdditionFilter

class ZeroAdditionFilter extends Filter
{
    //inputs:
    int mInitFeedbackData;
    int mShiftRegisterInput;
    //outputs
    //output 1: mInitFeedbackData
    int mAdditionOutput;


    public void init()
    {
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 2);
    }

    public void work()
    {
	mInitFeedbackData = input.popInt();
	mShiftRegisterInput = input.popInt();
	
	 
	output.pushInt(mInitFeedbackData);
	output.pushInt(mShiftRegisterInput);
    }

}//ZeroAdditionFilter  //i.e. the identity filter
    
class ShiftRegisterFilter extends Filter
{
    
    //private register variable:
    int mRegisterNumber;
    int mRegisterContents;
    //inputs:
    int mInitFeedbackData;
    int mDataStream;
    //output:
    int mRegisterOutput;
    int mTerminationCounter;
    
    public ShiftRegisterFilter(int number)
    {
        super(number);
    }
    public void init(int number)
    {
	mRegisterNumber = number;
	mTerminationCounter = 0;
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 2);
	mRegisterContents = 0; //initial value
    }

    public void work()
    {	
	mInitFeedbackData = input.popInt();
	mDataStream = input.popInt();
	mRegisterOutput = mRegisterContents;   //move datastream value into Shift Register
	mRegisterContents = mDataStream;       //move original contents to output
	mTerminationCounter++;
	/*
	if(mTerminationCounter < EncoderInput.mInputLength + 1)
	{
	    System.err.println("Shift Register Contents of Filter " + mRegisterNumber + " at iteration " + mTerminationCounter + " is " + mRegisterContents);
	    System.err.println("Shift Register Output of Filter " + mRegisterNumber + " at iteration " + mTerminationCounter + " is " + mRegisterOutput);
	    
	}
	*/
	output.pushInt(mInitFeedbackData);
	output.pushInt(mRegisterOutput);
    }
}//ShiftRegisterFilter

class CrcInputFilter extends Filter
{
    // int[] mdata;
    boolean counter;
    // EncoderInput mencodefile;

    public void init()
    {
	counter = false;
	// mencodefile = new EncoderInput();
	// mdata = new int[mencodefile.mInputLength];
	output = new Channel(Integer.TYPE, 1);
    }

    public void work()
    {
	if(!counter)
	{
	    counter = true;
	    output.pushInt(0);
	}
	else
	{
	    output.pushInt(1);
	}     
    }
	//mdata = mencodefile.readFile();
	   //for (int i = 0; i < mencodefile.mInputLength; i++)
	   //{
	    //System.errprintln("inputting " + mdata[i]);
    //output.pushInt(mdata[i]);
       //}
    

}//CrcInputFilter

class FeedbackEndFilter extends Filter
{
    int mRegisterOutput;
    int mInitFeedbackData;

    public void init()
    {
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 1);
    }
    
    public void work()
    {
	mInitFeedbackData = input.popInt();
	mRegisterOutput = input.popInt();
	//System.err.println("Shift Register 3 is " + mRegisterOutput);
	output.pushInt(mRegisterOutput);
    }
}//FeedbackEndFilter

class IntPrinter extends Filter {
    public void init()
    {
	input = new Channel(Integer.TYPE, 1);
    }
    
    public void work()
    {
	System.out.println(input.popInt());
    }
}

class CrcFeedbackLoop extends FeedbackLoop
{
    public void init()
    {
	this.setDelay(1);
	this.setJoiner(WEIGHTED_ROUND_ROBIN (1, 1));  //sequence: inputdata, fbdata
	this.setBody(new FeedbackBodyStream());
	this.setSplitter(DUPLICATE ());
	this.setLoop(new Identity(Integer.TYPE));
    }

    public int initPathInt(int index)
    {
	return 0;
    }
}//CrcFeedbackLoop

class FeedbackBodyStream extends Pipeline
{
    public void init()
    {
	this.add(new InitialAdditionFilter());
	this.add(new ShiftRegisterFilter(1));
	this.add(new AdditionFilter());
	this.add(new ShiftRegisterFilter(2));
	this.add(new AdditionFilter()); //x^2
	this.add(new ShiftRegisterFilter(3));
	//x^3
	this.add(new ShiftRegisterFilter(4));
	this.add(new AdditionFilter()); //x^4
	this.add(new ShiftRegisterFilter(5));
	this.add(new AdditionFilter()); //x^5
	this.add(new ShiftRegisterFilter(6));
	//x^6
	this.add(new ShiftRegisterFilter(7));
	this.add(new AdditionFilter()); //x^7
	this.add(new ShiftRegisterFilter(8));
	this.add(new AdditionFilter()); //x^8
	this.add(new ShiftRegisterFilter(9));
	//x^9
	this.add(new ShiftRegisterFilter(10));
	this.add(new AdditionFilter()); //x^10
	this.add(new ShiftRegisterFilter(11));
	this.add(new AdditionFilter()); //x^11
	this.add(new ShiftRegisterFilter(12));
	this.add(new AdditionFilter()); //x^12
	this.add(new ShiftRegisterFilter(13));
	//x^13
	this.add(new ShiftRegisterFilter(14));
	//x^14
	this.add(new ShiftRegisterFilter(15));
	//x^15
	this.add(new ShiftRegisterFilter(16));
	this.add(new AdditionFilter()); //x^16
	this.add(new ShiftRegisterFilter(17));
	//x^17
	this.add(new ShiftRegisterFilter(18));
	//x^18
	this.add(new ShiftRegisterFilter(19));
	//x^19
	this.add(new ShiftRegisterFilter(20));
	//x^20
	this.add(new ShiftRegisterFilter(21));
	//x^21
	this.add(new ShiftRegisterFilter(22));
	this.add(new AdditionFilter()); //x^22
	this.add(new ShiftRegisterFilter(23));
	this.add(new AdditionFilter()); //x^23
	this.add(new ShiftRegisterFilter(24));
	//x^24
	this.add(new ShiftRegisterFilter(25));
	//x^25
	this.add(new ShiftRegisterFilter(26));
	this.add(new AdditionFilter()); //x^26
	this.add(new ShiftRegisterFilter(27));
	//x^27
	this.add(new ShiftRegisterFilter(28));
	//x^28
	this.add(new ShiftRegisterFilter(29));
	//x^29
	this.add(new ShiftRegisterFilter(30));
	//x^30
	this.add(new ShiftRegisterFilter(31));
	//x^31
	this.add(new Identity(Integer.TYPE)); 

	//this.add(new ShiftRegisterFilter(3));
	this.add(new FeedbackEndFilter());
    }
   
}//FeedbackBodyStream

public class CrcEncoder32Test extends StreamIt
{
    public static void main(String args[])
    {
	new CrcEncoder32Test().run(args);
    }

    public void init()
    {
	this.add(new CrcInputFilter());
	this.add(new CrcFeedbackLoop());
	this.add(new IntPrinter());
    }
}//class CrcEncoder32Test





