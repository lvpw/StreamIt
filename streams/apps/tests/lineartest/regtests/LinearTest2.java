import streamit.library.*;

/**
 * Simple StreamIt application that contains some linear filters
 * that we can hopefully identify with the linear analysis pass
 * that I am going to be writing.
 **/
public class LinearTest2 extends StreamIt {
    public static void main(String[] args) {
	StreamIt app = new LinearTest2();
	app.run(args);
    }
    public void init() {
	this.add(new Source());
	this.add(new FloatIdentity());
	this.add(new FloatFilter());
	this.add(new FloatFilter2());
	this.add(new FloatFilter3());
	this.add(new FloatFilter4());
	this.add(new Sink());
    }
}


/**
 * Single, monotonically increasing source of floating point numbers.
 **/
class Source extends Filter {
    float INCREMENT = (float).01;
    float currentValue = 0;
    public void init() {
	output = new Channel(Float.TYPE, 1);
	this.currentValue = 0;
    }
    public void work() {
	output.pushFloat(this.currentValue);
	this.currentValue += INCREMENT;
    }
}

/**
 * Simple float sink that prints what it gets.
 **/
class Sink extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 1);
    }
    public void work() {
	System.out.println(input.popFloat());
    }
}

/**
 * About the simplest filter you can get that computes a linear function
 **/
class FloatIdentity extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 1);
	output = new Channel(Float.TYPE, 1);
    }
    public void work() {
	output.pushFloat(input.popFloat());
    }
}

/**
 * This is the filter which contains the work function that
 * I used to test the linear analysis.
 **/
class FloatFilter extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 3);
	output = new Channel(Float.TYPE, 3);
    }
    public void work() {
	// should result in 3rd col: [1,1,1] + 0
	output.pushFloat(input.peekFloat(0) +
			 input.peekFloat(1) +
			 input.peekFloat(2));

	// should result in 2nd col: [3,2,10] + 7
	output.pushFloat(1 +
			 2*input.peekFloat(1) +
			 input.peekFloat(2)*3 +
			 2*3 +
			 2*input.peekFloat(0)*5);

	// should result in 1st col: [0,0,0] + -11	
	output.pushFloat(-11);

	// pop off the values from the input tape
	input.popFloat();
	input.popFloat();
	input.popFloat();
    }
}

/**
 * Use this class to make sure that peeks work correctly
 * when interspersed with pops.
 **/
class FloatFilter2 extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 4);
	output = new Channel(Float.TYPE, 3);
    }
    public void work() {
	// ok, first make a nice, normal column with peek expressions
	// result in col 3 being: [8, 5, 6, 12] + 3
	output.pushFloat(input.peekFloat(0) * 2 * 1 * 3 * 1 * 2 +
			 3 * 1 * 1 * 2 * input.peekFloat(1) +
			 5 * input.peekFloat(2) +
			 input.peekFloat(3) * 8 +
			 2+4+2+1-3-3);

	// pop off a result
	input.popFloat();

	// now, push using updated peek values and ensure that they are correct
	// result in col 2 being [4 3 2 0] + 7
	output.pushFloat(input.peekFloat(0) * 2 +
			 input.peekFloat(1) * 3 +
			 input.peekFloat(2) * 4 +
			 8-1);

	// pop again and figure out what is going on
	input.popFloat();

	// col 1 = [1 1 0 0] + 2
	output.pushFloat(input.peekFloat(0) +
			 input.peekFloat(1) +
			 2);

	// pop off the last 2 items
	input.popFloat();
	input.popFloat();
    }
}

/**
 * Use this class to test division
 **/
class FloatFilter3 extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 4);
	output = new Channel(Float.TYPE, 1);
    }
    public void work() {
	// col 1 = [1/5, 2/3, 11, 3/4] + 21/32
	output.pushFloat(input.peekFloat(3) / 5 +
			 (2*input.peekFloat(2)) / 3 +
			 11 * input.peekFloat(1) +
			 (3 * input.peekFloat(0)) / (2 * 2 * 2 / 2) +
			 ((float)21)/((float)32));
	input.popFloat();
	input.popFloat();
	input.popFloat();
	input.popFloat();
    }
}


/**
 * Use this one to make sure that we don't screw and do bad things with division
 **/
class FloatFilter4 extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 1);
	output = new Channel(Float.TYPE, 1);
    }
    public void work() {
	// non linear
	output.pushFloat(1/input.peekFloat(0));
	input.popFloat();
    }
}


