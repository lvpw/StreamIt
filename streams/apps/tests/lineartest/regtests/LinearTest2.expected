FloatFilter2_1.matrix
[[1.0+0.0i 4.0+0.0i 8.0+0.0i]
 [1.0+0.0i 3.0+0.0i 5.0+0.0i]
 [0.0+0.0i 2.0+0.0i 6.0+0.0i]
 [0.0+0.0i 0.0+0.0i 12.0+0.0i]]
FloatFilter2_1.vector
[[2.0+0.0i 7.0+0.0i 3.0+0.0i]]
FloatFilter3_1.matrix
[[0.2+0.0i]
 [0.6666666666666666+0.0i]
 [11.0+0.0i]
 [0.75+0.0i]]
FloatFilter3_1.vector
[[0.65625+0.0i]]
FloatFilter_1.matrix
[[0.0+0.0i 3.0+0.0i 1.0+0.0i]
 [0.0+0.0i 2.0+0.0i 1.0+0.0i]
 [0.0+0.0i 10.0+0.0i 1.0+0.0i]]
FloatFilter_1.vector
[[-11.0+0.0i 7.0+0.0i 0.0+0.0i]]
FloatIdentity_1.matrix
[[1.0+0.0i]]
FloatIdentity_1.vector
[[0.0+0.0i]]
LinearWrapper_1.matrix
[[238.93333333333334+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [181.73333333333332+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [639.3333333333334+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [3.0+0.0i 22.333333333333332+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [2.25+0.0i 16.2+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [8.25+0.0i 65.26666666666667+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 38.0+0.0i 1.6+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 30.0+0.0i 1.6+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 94.0+0.0i 1.6+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 4.5+0.0i 6.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 3.0+0.0i 4.800000000000001+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 15.0+0.0i 14.4+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 138.75+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 101.33333333333333+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 400.6666666666667+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 329.8+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 238.93333333333334+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 965.8666666666667+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 396.75+0.0i 6.133333333333333+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 264.75+0.0i 6.133333333333333+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 1320.75+0.0i 6.133333333333333+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 21.2+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 16.8+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 52.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 41.25+0.0i 4.2+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 28.0+0.0i 3.2+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 134.0+0.0i 11.200000000000001+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 189.41666666666666+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 138.75+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 544.0833333333333+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 27.0+0.0i 90.86666666666666+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 18.0+0.0i 90.86666666666666+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 90.0+0.0i 90.86666666666666+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 334.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 266.6666666666667+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 805.3333333333334+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 2.25+0.0i 16.2+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 1.5+0.0i 12.266666666666666+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 7.5+0.0i 43.73333333333333+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 55.25+0.0i 5.800000000000001+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 41.25+0.0i 4.2+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 153.25+0.0i 17.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 57.86666666666667+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 55.46666666666667+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 74.66666666666667+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 88.5+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 62.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 274.0+0.0i]]
LinearWrapper_1.vector
[[-2005.6604166666666+0.0i -53.41041666666667+0.0i -315.17708333333337+0.0i 851.3729166666667+0.0i -42.41041666666666+0.0i 171.00625+0.0i -120.64375+0.0i -15.127083333333331+0.0i -129.44375+0.0i]]
